quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -364,6 +369,52 @@ void setDefaultUserProfileConfig(KeycloakRecorder recorder, UserProfileBuildItem
         recorder.setDefaultUserProfileConfiguration(configuration.getDefaultConfig());
     }
 
+    /**
+     * Get datasource name obtained from the persistence.xml file based on this order:
+     * <ol>
+     *      <li> return {@link JdbcSettings#JAKARTA_JTA_DATASOURCE} or {@link JdbcSettings#JAKARTA_NON_JTA_DATASOURCE} if specified
+     *      <li> return {@link AvailableSettings#DATASOURCE} property if specified
+     *      <li> return persistence unit name
+     * </ol>
+     * Can be removed after removing support for persistence.xml files
+     */
+    private static String getDatasourceNameFromPersistenceXml(PersistenceUnitDescriptor descriptor) {
+        if (descriptor == null) {
+            throw new IllegalStateException("Descriptor cannot be null");
+        }
+        final BiConsumer<String, String> infoAboutUsedSourceForDsName = (source, name) -> logger.infof(
+                "Datasource name '%s' is obtained from the '%s' configuration property in persistence.xml file. " +
+                        "Use '%s' name for datasource options like 'db-kind-%s'.", name, source, name, name);
+
+        String persistenceUnitName = descriptor.getName();
+        Properties properties = descriptor.getProperties();
+
+        // 1. return Jakarta properties
+        @SuppressWarnings("removal") // Once we remove persistence.xml support, we can remove it as well
+        var jakartaPropertyName = switch (descriptor.getTransactionType()) {
+            case JTA -> JAKARTA_JTA_DATASOURCE;
+            case RESOURCE_LOCAL -> JAKARTA_NON_JTA_DATASOURCE;
```

Comment:
This looks good overall, but I don't know if we should support RESOURCE_LOCAL as it doesn't seem to be well supported in quarkus (https://github.com/quarkusio/quarkus/issues/2628) and moving forward to the hibernate options there's no replacement for that. It will always be JTA.